src/coprocessor/endpoint.rs

```diff
@@ -392,12 +396,27 @@ impl<E: Engine> Endpoint<E> {
         let handle_request_future = check_deadline(handler.handle_request(), deadline);
         let handle_request_future = track(handle_request_future, &mut tracker);
 
-        let deadline_res = if let Some(semaphore) = &semaphore {
-            limit_concurrency(handle_request_future, semaphore, LIGHT_TASK_THRESHOLD).await
+        let result = if let Some(semaphore) = &semaphore {
+            limit_concurrency(
+                handle_request_future,
+                semaphore,
+                semaphore_waiter_limit,
+                LIGHT_TASK_THRESHOLD,
+            )
+            .await
+            .and_then(|deadline_res| deadline_res.map_err(Error::from).and_then(|res| res))
         } else {
-            handle_request_future.await
+            handle_request_future
+                .await
+                .map_err(Error::from)
+                .and_then(|res| res)
         };
-        let result = deadline_res.map_err(Error::from).and_then(|res| res);
+        if matches!(
+            result,
+            Err(Error::DeadlineExceeded | Error::MaxPendingTasksExceeded)
```

Comment:
I think only `Error::DeadlineExceeded` needs to be printed.

Reply:
Do you think we just revert back the slow log behavior, print the slow log when `req_lifetime` > threshold?

Reply:
If the task takes too long and be aborted, will it still show in the slow log?

Reply:
If we use `req_lifetime` to check, it will.